### PR TITLE
Add the `packaged` sigil to files

### PR DESCRIPTION
**Pull request by elliottt**
Like the `compiled` sigil this is Stripe-specific, but will help with landing some improvements to the performance of the current packages implementation.

### Motivation

Pre-work for the package visibility checker.

### Test plan

See included automated tests.

### CODE REVIEW

File: core/Files.h [hunk 1]
```diff
@@ -126,11 +131,14 @@ class File final {
...
     const CompiledLevel compiledLevel;
+    const PackagedLevel packagedLevel;
 
 private:
     std::shared_ptr<const FileHash> hash_;
 };
-CheckSize(File, 96, 8);
+
+// TODO: see if we can pack `packagedLevel` in a bit better to get this back down to 96
+CheckSize(File, 104, 8);
```
Comment (elliottt) [inline core/Files.h:141]: A fresh challenge for you, @froydnj :D

Comment (froydnj) [reply core/Files.h:141]: There should be two spare bytes right after `flags` which will accommodate this. :D

**Review by froydnj — APPROVED** [review]
Is there some documentation being added in a follow up package about what the semantics are?

**Comment by elliottt** [comment]
> Is there some documentation being added in a follow up package about what the semantics are?

Yes, with this PR the sigil is unused.

**Pull request merged**